IntelFsp2Pkg/Library/BaseFspSwitchStackLib/Ia32/Stack.nasm

```diff
@@ -61,6 +86,103 @@ ASM_PFX(FspSwitchStack):
     ; Restore previous contexts
     lidt    [esp]
     add     esp, 8
+
+    cmp     byte [dword ASM_PFX(FeaturePcdGet (PcdFspSaveRestorePageTableEnable))], 0
+    jz      SkipPagetableRestore
+    ; [esp]    restores new cr0
+    ; [esp+4]  restores new cr3
+    ; [esp+8]  restores new cr4
+    ; [esp+12] restores new Efer
+
+    ;
+    ; If NXE bit is changed to 1, change NXE before CR register
+    ; This is because Nx bit in page table entry in new CR3 will be invalid
+    ; if updating CR3 before EFER MSR.
+    ;

```

Comment:
When new EFER.NXE == 1, the restore flow is:    EFER --> CRx
Otherwise:       CRx --> EFER

When new CR0.PG == 1, the restore flow for CRx is: CR3 --> CR4 --> CR0
Otherwise, the restore flow is: CR0 --> CR3 --> CR4